src/scripts/main.js

```diff
@@ -1,3 +1,306 @@
+
+    formValuesArr.forEach(([key, value]) => {
+      const employeeField = newEmployee.insertCell();
+
+      if (key !== 'salary') {
+        employeeField.innerText = value;
+      } else {
+        employeeField.innerText = toCurrencyFormat(value);
+      }
+    });
+
+    form.reset();
+
+    pushNotification(0, 0, 'Congratulations!',
+      'Employee added to the table.', 'success');
+  }
+});
+
+function toCurrencyFormat(number) {
+  return parseInt(number)
+    .toLocaleString('en-US', {
+      style: 'currency', currency: 'USD',
+    }).replace('.00', '');
+}
+
+function validateForm({ name: validName, age: validAge }) {
+  if (validName.length < 4) {
+    pushNotification(
+      0, 0, 'Enter correct name',
+      'The name should be at least 4 characters long', 'error'
+    );
+
+    return false;
+  }
+
+  if (validAge < 18 || validAge > 90) {
+    pushNotification(
+      0, 0, 'Age restrictions',
+      'Sorry, your age does not meet our requirements', 'error'
+    );
+
+    return false;
+  }
+
+  return true;
+}
+
+function pushNotification(posTop, posRight, title, description, type) {
+  const notification = document.createElement('div');
+
+  notification.classList.add('notification', type);
+  notification.style.top = `${posTop}px`;
+  notification.style.right = `${posRight}px`;
+  notification.dataset.qa = 'notification';
+
+  notification.innerHTML = `
+    <h2 class="title">${title}</h2>
+    <p>${description}</p>
+  `;
+
+  document.body.append(notification);
+
+  setTimeout(() => notification.remove(), 4000);
+};
+
+// Implementation of editing of table cells by double-clicking on it
+
+tableBody.addEventListener('dblclick', e => {
+  const selectedCell = e.target.closest('td');
+  const initValue = selectedCell.innerText;
+  const index = selectedCell.cellIndex;
+
+  selectedCell.innerText = '';
+
+  switch (index) {
+    case 2:
+      selectedCell.innerHTML = `
+        <select class="cell-input" value="${initValue}">
+          ${officeOptions.map(office => `
+            <option value="${office}">${office}</option>
+          `).join('')}
+        </select>
+      `;
+
+      selectedCell.querySelectorAll('option')
+        .forEach(option => {
+          if (option.innerText === initValue) {
+            option.setAttribute('selected', 'selected');
+          }
+        });
+      break;
+
+    case 3:
+      selectedCell.innerHTML = `
+        <input
+          class="cell-input"
+          type="number"
+          value="${initValue}"
+          min="18"
+          max="90"
+        >
+      `;
+      break;
+
+    case 4:
+      selectedCell.innerHTML = `
+        <input
+          class="cell-input"
+          type="number"
+          value="${toFormat(initValue)}"
+        >
+      `;
+      break;
+
+    default:
+      selectedCell.innerHTML = `
+        <input
+          class="cell-input"
+          type="text"
+          pattern="[a-zA-Z0-9]+[a-zA-Z0-9 ]+"{4}
```

Comment:
Remove the '{4}' from the pattern attribute in the input element, as it doesn't seem to have any effect and may cause confusion. Use minLength attribute instead.